### PR TITLE
Add GoodScaler class that derives KeepArScaler

### PR DESCRIPTION
**Pull request by shssoichiro**

### CODE REVIEW

**Comment by shssoichiro** [comment]
@LightArrowsEXE do you think this is sufficient for what we were discussing in Discord?

File: soifunc/resize.py
Comment (LightArrowsEXE) [inline soifunc/resize.py:22]: You probably want to somehow allow the user to pass on kwargs to at least Spline36 (assuming you want stuff like `keep_ar` to function. 

Consider something like this instead:

```py
def __init__(self, scaler: ScalerT = Spline36) -> None:
    self.scaler = HybridScaler(SSIM, Spline36.from_param(scaler))
```

File: soifunc/resize.py
Comment (LightArrowsEXE) [inline soifunc/resize.py:18]: Might wanna also inherit `vskernels.KeepArScaler` and copy over the code for `scale` if you want to add `keep_ar` support like what we brought up on Discord.

**Review by LightArrowsEXE — CHANGES_REQUESTED** [review]
Please test everything I suggested to make sure it works properly before you merge the PR!

If you're just wondering whether this will work, I believe it will (other than my review comment about the missing return statement).

EDIT: Though I also have to ask: why make this a class, as opposed to having users use `HybridScaler.scale()`? That seems more intuitive than basically creating a class that does what `HybridScaler`, from what I remember, already does?

File: soifunc/resize.py
Comment (LightArrowsEXE) [inline soifunc/resize.py:14]: You need to also export GoodScaler!

File: soifunc/resize.py
Comment (shssoichiro) [reply soifunc/resize.py:57]: I'm so used to using Rust with its implicit returns that I keep forgetting them in Python :facepalm:

**Comment by shssoichiro** [comment]
> Though I also have to ask: why make this a class, as opposed to having users use HybridScaler.scale() and implement some of the logic here in there instead?

HybridScaler is really just supposed to be a helper class, though it could be used on its own. The purpose of GoodResize was to use an opinionated set of scalers that are intended to provide the "best" (subjective) quality currently available (without targeting any specific content type, i.e. so not using Waifu2x). And to keep that call as simple as possible. I want to still be able to just call `GoodScaler.scale(clip, 1920, 1080)` without _needing_ to specify any other parameters.

**Comment by shssoichiro** [comment]
This seems to work based on local testing, though it is blocked on https://github.com/Jaded-Encoding-Thaumaturgy/vs-scale/pull/85

**Pull request merged**
